scripts/check_doc_gate.py

```
@@ -281,11 +284,22 @@ def evaluate_rules(
-    result = subprocess.run(
-        ["git", *args], cwd=REPO_ROOT, capture_output=True, text=True, check=True,
-    )
-    return result.stdout
+class GitCommandError(Exception):
+    """Raised when a git command fails, so infrastructure failures are
+    distinguishable from genuine doc-gate violations."""
+
+
+def _run_git(args: list[str], ref: str | None = None) -> str:
+    try:
+        result = subprocess.run(
```

Ruff (0.16.1)

[error] `subprocess` call: check for execution of untrusted input

(S603)